app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -46,19 +47,40 @@ abstract class OfferViewModel : ViewModel() {
 }
 
 class OfferViewModelImpl(
-    private val offerRepository: OfferRepository
+    _quoteIds: List<String>,
+    private val offerRepository: OfferRepository,
 ) : OfferViewModel() {
 
+    private lateinit var quoteIds: List<String>
+
     override val autoStartToken = MutableLiveData<SignOfferMutation.Data>()
     override val signStatus = MutableLiveData<SignStatusFragment>()
     override val signError = MutableLiveData<Boolean>()
 
     init {
-        load()
+        if (_quoteIds.isEmpty()) {
+            viewModelScope.launch {
+                val idResult = offerRepository.quoteIdOfLastQuoteOfMember().safeQuery()
+                if (idResult !is QueryResult.Success) {
```

Comment:
Can we move this OfferRepository (something like `GetAddressChangeStoryIdUseCase`) and return our own Result class, and then map the result to a viewstate? Perhaps thats the scope of another PR :)

Reply:
I gave it a shot in a new commit, check it out!